text/0314-cdk-construct-hub.md

```diff
@@ -0,0 +1,665 @@
+npmjs.com, Maven Central, PyPI, NuGet, etc.
+
+## Specification
+
+### Landing Page
+
+> [Example rendering](#rendering-landing-page)
+
+The landing page for the Construct Hub includes a search bar that allows
+developers to search for constructs that solve their use-case using relevant
+keywords, and optionally filtering on a specific programming language and
+constructs domain (AWS CDK / CDK8s / CDKtf / Projen / ...).
+
+It may also feature a short-list of packages, selected based on popularity or
+latest release date. The format of that list is the same as the format of search
+results.
+
+### Search Results Page
+
+When a search is performed using the search bar in the landing page (or in the
+header of all other pages), a list of relevant results is displayed as tiles,
+similar to the short-list of packages shown on the landing page.
+
+The results grid includes the following elements about returned packages:
+
+- Name
+- Short description (truncated to fit if needed)
+- Author name
+- Publish date
+- Latest version
+
+In the future, it may also include quality or popularity indications such as
+the number of downloads of that package.
+
+### Package Detail Page
+
+> [Example rendering](#rendering-package-detail-page)
+
+Clicking on a given package's tile in the landing page or search result page
+brings the user to a package detail page that represents the following
+information:
+
+- Package name, tags, and description
+- Overview of publishing information (author, version, license, etc...)
+  - Where possible, links to the source repository, issue tracker, etc... based
+    on information registered in the `.jsii` assembly file.
+- Install instructions
+- Available languages
+- Package's documentation
+  - Rendered content of the package's `README.md`
+  - API reference documentation, in all supported languages (only the version
+    for the currently selected language is displayed)
+
+The list of available languages is also a selector: upon clicking a language,
+the installation instructions and documentation automatically switch to the
+relevant form for the currently selected language.
+
+### Support
+
+The Construct Hub offers ways for customers to engage maintainers about
+various problems, including the following (actually routed to GitHub issue
+templates):
+
+- A malicious package was identified
+- Incorrectly rendered documentation
+- Package un-listing request
+- Offensive content
+
+When a contact request originates from a package detail page, the customer is
+guided to the package's issue tracker or public repository if their issue is
+with the package itself (and not about it's listing in Construct Hub). The
+routing is based on metadata configured on the package itself, through the bug
+tracker URL, repository URL, and homepage URL.
+
+## Technical Design
+
+### Overview
+
+![Architecture Diagram](../images/cdk-construct-hub/architecture-diagram.png)
+
+### What is the high level implementation plan?
+
+#### Overview
```

Comment:
Worth adding some design tenets. One important one is that v1 will be solely based on cloudfront as the backend so there are no "write" routes on the server (not sure how to articulate it but it's a really important one).

Analytics will be implemented through a 3rd party (Adobe analytics)